pubsub/iterator_test.go

```diff
@@ -618,3 +619,207 @@ func TestPingStreamAckDeadline(t *testing.T) {
 	}
 	iter.eoMu.RUnlock()
 }
+
+func compareCompletedRetryLengths(t *testing.T, completed, retry []*AckResult, wantCompleted, wantRetry int) {
```

Comment:
Not related to this PR but at L186 a testing.T is sent to another goroutine and calls fatal which will cause a panic.